build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/biomeTasks.ts

```diff
@@ -3,14 +3,46 @@
  * Licensed under the MIT License.
  */
 
+import { readFile } from "fs/promises";
 import path from "path";
+import ignore from "ignore";
+import * as JSON5 from "json5";
+import { merge } from "ts-deepmerge";
 import { getResolvedFluidRoot } from "../../../common/fluidUtils";
 import { GitRepo } from "../../../common/gitRepo";
 import { LeafWithFileStatDoneFileTask } from "./leafTask";
 
 // switch to regular import once building ESM
 const findUp = import("find-up");
 
+/**
+ * A type defining the Biome config file. Only the fields we care about are included.
+ *
+ * @privateRemarks
+ *
+ * Biome does have a JSON schema for the config format, so in the future if we need more comprehensive config file
+ * handling, this should be updated to use the schema instead.
+ */
+interface BiomeConfig {
+	extends?: string[];
+	files?: {
+		ignore?: string[];
+	};
+	formatter?: {
+		ignore?: string[];
+	};
+}
+
+/**
+ * This task enables incremental build support for Biome formatting tasks. It reads Biome configuration files to load
+ * ignore settings, and will not consider ignored files when checking if the task needs to run.
+ *
+ * In addition, files that are ignored by git will be excluded. Internally the task uses git itself to enumerate files.
+ *
+ * IMPORTANT: While ignore settings are loaded and applied from the Biome configuration file, the "include" settings are
+ * not consulted. This means that files may not be properly excluded by the task when using "include" and "ignore"
+ * together.
```

Comment:
This limitation is OK for our current usage, but we should plan to add this capability in a follow up if not now.